File: .leetcode/455.分发饼干.cpp
/*
 * @lc app=leetcode.cn id=455 lang=cpp
 *
 * [455] 分发饼干
 */

// @lc code=start
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int j = g.size() - 1;
        int ans = 0;
        for(int i = s.size()-1;i>=0;i--){
            for(j;j>=0;j--){
                if(s[i]>=g[j]){
                    ans++;
                    j--;
                    break;
                }
            }
        }
        return ans;
    }
};
// @lc code=end

File: .leetcode/455.分发饼干.java
import java.util.Arrays;

/*
 * @lc app=leetcode.cn id=455 lang=java
 *
 * [455] 分发饼干
 */

// @lc code=start
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        if(s.length == 0)return 0;
        Arrays.sort(g);
        Arrays.sort(s);
        int index = 0;
        for(int i = 0;i < s.length;i++){
            if(index < g.length && s[i] >= g[index]){
                index++;
            }
        }
        return index;
    }
}
// @lc code=end

